### PR TITLE
ARROW-12901: [R] Follow on to more examples

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
WRT @jonkeane 's questions on the JIRA ticket:
> https://github.com/apache/arrow/pull/10368/files#diff-9fe1a055f9a4df7088de4e0c759c20138422b8bdb9bba04ed74c7881aa787019R66-R67
> Should this use @examplesIf ?

Yep - have updated now!

> https://github.com/apache/arrow/pull/10368/files#diff-0229d62b9653ff7124e6e2f202e51d46d4d8fcaf14baf83ccf8c047c7c344579R78-R80
> These are both genuine questions, I'm not sure we want to include either or both of them:
>   * This might be too much indirection, but would it be better/easier to use `write_dataset() here?
Nice, that's made that code a LOT nicer to look at.
>   * Do we want to talk about reading a partitioned dataset here?
Yeah, I think this is probably a good place to put that - what kinds of things do you think we should mention?

**Comment by thisisnic** [comment]
@jonkeane - how does this look now?

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-r-without-arrow

File: r/R/dataset.R [hunk 1]
```diff
@@ -69,30 +69,52 @@
...
+#' write_dataset(data, tf2, format = "ipc")
 #' # This line will results in errors when you try to work with the data
-#' \dontrun{open_dataset(c(file.path(tf, "file1.csv"), file.path(tf, "file2.csv")))}
-#' # This is the correct way to open a dataset containing CSVs
-#' open_dataset(c(file.path(tf, "file1.csv"), file.path(tf, "file2.csv")), format = "csv") 
+#' \dontrun{open_dataset(tf2)}
+#' # This line will work
+#' open_dataset(tf2, format = "ipc") 
+#' 
+#' ## You can specify file partitioning to include it as a field in your dataset
```
Comment (jonkeane) [inline r/R/dataset.R:98]: is the double-# intentional here?

File: r/R/dataset.R [hunk 2]
```diff
@@ -69,30 +69,52 @@
...
+#' 
+#' # With no partitioning specified, dataset contains all files but doesn't include
+#' # directory names as field names
+#' open_dataset(tf3)
+#' 
+#' # Create a Schema which specifies the field names for the folders, in order
+#' md_schema <- schema(Month = int8(), Day = int8())
+#' 
+#' # Now that partitioning has been specified, your dataset contains columns for Month and Day
+#' open_dataset(tf3, partitioning = md_schema)
```
Comment (jonkeane) [inline r/R/dataset.R:117]: Could we also include the simpler version of this too?

`open_dataset(tf3, partitioning = c("Month", "Day"))`

**Review by jonkeane — COMMENTED** [review]
This is looking good. A few more suggestions / comments now that I'm seeing it written out. We should also probably wait for #10455 to merge before we send this one so we can rebase + run the tests here and make sure that's all sorted.

Additionally, I noticed that a lot of this content is in the long running + starting/stoping PR #9748 as well. The only thing that isn't here that is in #9748 is some reference to how `df %>% group_by("foo", "bar") %>% write_dataset(...)` is the same thing as `write_dataset(df, partitioning = c("foo", "bar"))` maybe we could add an example or two of that and then close there other PR?

**Comment by thisisnic** [comment]
> This is looking good. A few more suggestions / comments now that I'm seeing it written out. We should also probably wait for #10455 to merge before we send this one so we can rebase + run the tests here and make sure that's all sorted.
> 
> Additionally, I noticed that a lot of this content is in the long running + starting/stoping PR #9748 as well. The only thing that isn't here that is in #9748 is some reference to how `df %>% group_by("foo", "bar") %>% write_dataset(...)` is the same thing as `write_dataset(df, partitioning = c("foo", "bar"))` maybe we could add an example or two of that and then close there other PR?

That PR is for `write_dataset`, whereas this one is for `open_dataset` though.  Maybe we could think more about what goes in which and get them more in sync?

File: r/R/dataset.R (hunk 1 above)
Comment (thisisnic) [reply r/R/dataset.R:98]: Yep, I was trying to use it to specify a new section - however, if this isn't widely done, maybe I should do it differently?

File: r/R/dataset.R (hunk 2 above)
Comment (thisisnic) [reply r/R/dataset.R:117]: Yep

**Comment by jonkeane** [comment]
> That PR is for write_dataset, whereas this one is for open_dataset though. Maybe we could think more about what goes in which and get them more in sync?

Ah, right of course. Yeah, it might be good to harmonize them (or maybe even combine them into one page?) But regardless we can do that on a follow on (or on that other ticket), we don't need to do that in this one.

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-r-without-arrow

**Comment by jonkeane** [comment]
@github-actions crossbow submit test-r-minimal-build

File: r/R/dataset.R (hunk 1 above)
Comment (jonkeane) [reply r/R/dataset.R:110]: We can keep it there — I don't think there's a standard pattern either in the package or elsewhere (that I know of, but let me know if you know of the state of art in other packages!)

**Comment by jonkeane** [comment]
This looks good, I've run the two builds we had issues with in the past. This shouldn't be necessary, but just in case I'll also run `crossbow -g r` I don't think there are other builds that will be impacted by it, but just in case! If that goes fine I'll merge. Thanks!

**Comment by jonkeane** [comment]
@github-actions crossbow submit -g r

**Pull request closed without merging**